### PR TITLE
Team colors fix

### PR DESCRIPTION
**Pull request by AACO**
Should fix team colors, only tested locally though, a few person MP test should really be executed before merging.

### CODE REVIEW

**Comment by PabstMirror** [comment]
I have no clue why this broke
event is still there in cba

BIKI says global/global, but I remember it being a little weird when I was testing 1.62

**Comment by AACO** [comment]
It looks like the event is registered postInit, while this can be called during init (the waitAndExecute _should_ push us past post init), but I'm guessing that it's some strange race issue.

File: addons/teamColors/functions/fnc_setColor.sqf [hunk 1]
```diff
@@ -15,8 +16,8 @@
...
 TRACE_1("Params",_this);
 
-[{
-    params ["_unit", "_fireteam"];
-    TRACE_2("setting fireteam",_unit,_fireteam);
-    ["CBA_teamColorChanged", [_unit, _fireteam], [_unit]] call CBA_fnc_targetEvent;
-}, _this, 1] call CBA_fnc_waitAndExecute;
+[
+    {(_this select 0) assignTeam (_this select 1);},
+    {!isNull player},
```
Comment (PabstMirror) [inline addons/teamColors/functions/fnc_setColor.sqf]: I think the order is reversed, `[condition,statement,args]`
But this also runs on dedi, so player will be null there

Comment (AACO) [inline addons/teamColors/functions/fnc_setColor.sqf]: Good call, I'm still playing with it a bit to do testing, assignTeam needs to be called when the player is finished initializing, so we might be better off checking unit locality then waiting for player initialization.

**Comment by AACO** [comment]
Seems to work on dedicated with AI, worth a try......

**Pull request merged**